### ISSUE TITLE
Linux assemblies

### ISSUE THREAD

**Issue by paulorenanmelo** [DemolitionStudios/DemolitionMediaHap#7]
Where should I ask for the assemblies for linux?

**Comment by FisheyeWout** [comment]
Given that this plugin is mostly used for non-consumer installations Linux support would be very useful!

**Comment by vollstock** [comment]
Indeed, being the maintainer of [Tooloop OS](https://github.com/vollstock/Tooloop-OS/), I too would love to see a Linux version of this :-)

**Comment by leavittx** [comment]
Anybody still interested in linux assemblies?
It's half finished already, I can make a release soon if there is still some interest!

**Comment by vollstock** [comment]
In general yes. We deploy all our public / kiosk installations on Linux.
Having a proper HAP video player for Unity in our Toolkit would be great.

**Comment by leavittx** [comment]
Great to hear! Thanks.
I'll work on Linux build of 2022 version for next 1-2 months, and release it after that.
Please let me know if the Pro Sync version also in the interest of anyone.

**Comment by vollstock** [comment]
We`re definitely interested. However given the steaper price we would only purchase this once a project demands it.

**Comment by leavittx** [comment]
Hi @vollstock and all
Just wanted to share the intermediate results.
I worked on building ffmpeg/native plugin for almost a month now (not full-time, but kinda a lot).
Along with getting the ffmpeg/plugin fully workable on my Ubuntu 22.04, several other general problems were solved, the main one is fixing the mid-size memory leak that was there on all platforms. Now the plugin should be suitable for 24/7 installations without restarting (the leak was about 1Mb/min). I'm now wrapping things up and preparing new asset for Linux. From my experience it usually takes up to 1-1.5 months to get a new asset approved. So if anybody wants to test it sooner, we can chat. Otherwise I expect it to be online on Asset Store somewhere in January :)

I didn't manage to get GDeflate compute shaders work on Linux unfortunately, however from my small research it's more Unity integration problem, as DXC compiler can compile them without problem on Linux. So I will submit a bug report after I submit all the assets

**Comment by leavittx** [comment]
Would targeting a glibc-2.35+ system be good enough? I.e. Ubuntu 22.04+ would work, but 20.04 won't. 
I can try to build with glibc statically otherwise, but that doesn't seem to be a very nice solution to me (though there are other solutions)
I can see that Tooloop OS is based on 22.04. Hoping everybody developing media stuff is on recent Linux versions :)

**Comment by vollstock** [comment]
Well yes, I am very bad at updating docs but try and keep Tooloop OS in sync with current LTS versions. So yes, fine fir me. Probably fine for most, thoufgh I can’t speak for others, od course.

**Comment by leavittx** [comment]
Hi
The asset got approved for the Asset Store!
Simple playback version: https://assetstore.unity.com/packages/tools/video/demolition-media-hap-2022-linux-268419
Pro Sync version: https://assetstore.unity.com/packages/tools/video/demolition-media-hap-pro-sync-linux-271017

**Comment by leavittx** [comment]
@vollstock @FisheyeWout @paulorenanmelo
